tests/common.rs

```diff
@@ -70,13 +99,15 @@ pub fn handle_test_error(
     }
 }
 
+
+#[cfg(feature = "rustls-tls")]
 #[macro_export]
-macro_rules! tester {
+macro_rules! rustls_tester {
     ($f:ident, $endpoint:expr) => {{
         use std::sync::{Arc, Mutex};
         use std::thread;
 
-        let c = common::select_client_type($endpoint);
+        let c = common::rustls_type($endpoint);
```

Comment:
Let's avoid the duplication here, and have the macro take `$connector:expr` instead of `$endpoint:expr`. And then make this line be
```rust
let c = $connector;
```
Then, rename this macro to `tester_inner`, and add another macro like this:
```rust
macro_rules! tester {
    ($f:ident, $endpoint:expr) => {{
        #[cfg(feature = "rustls-tls")]
        tester_inner!($f, rustls_type($endpoint));
        #[cfg(feature = "openssl-tls")]
        tester_inner!($f, opentls_type($endpoint));
}
```
That way, all the tests can stay the same, and they'll test both TLS implementations.